extras-scalapb/src/main/scala/com/avast/clients/rabbitmq/extras/format/ScalaPBAsJsonProductConverter.scala

```diff
@@ -0,0 +1,42 @@
+package com.avast.clients.rabbitmq.extras.format
+
+import com.avast.bytes.Bytes
+import com.avast.clients.rabbitmq.ProductConverter
+import com.avast.clients.rabbitmq.api.{ConversionException, MessageProperties}
+import scalapb.GeneratedMessage
+import scalapb.json4s.Printer
+
+import scala.annotation.implicitNotFound
+import scala.reflect.ClassTag
+import scala.util.control.NonFatal
+
+@implicitNotFound("Could not generate ScalaPBAsJsonProductConverter for ${A}, try to import or define some")
+trait ScalaPBAsJsonProductConverter[A <: GeneratedMessage] extends ProductConverter[A]
+
+object ScalaPBAsJsonProductConverter {
+  def derive[A <: GeneratedMessage: ScalaPBAsJsonProductConverter](): ScalaPBAsJsonProductConverter[A] =
+    implicitly[ScalaPBAsJsonProductConverter[A]]
+
+  // Printing is intentionally configured to use integer for enums
+  // because then the serialized data is valid even after values renaming.
+  private val jsonPrinter = new Printer().formattingEnumsAsNumber
```

Comment:
On its own, using integers for enums is a reasonable thing to do. But that's not what `grpc-json-bridge` does, so I would argue against it. IMHO it's very important to be compatible and interoperable with it.

Reply:
[The Protobuf-JSON definition](https://developers.google.com/protocol-buffers/docs/proto3#json) explicitly says following: `Parsers accept both enum names and integer values.`
So IMHO this is not violation of any interoperability.

Reply:
I'm not really worried about breaking the Protobuf-JSON specification. It's just that receiving different JSONs from `grpc-json-bridge` and from Rabbit which are supposed to be the same thing makes me a bit uncomfortable.
So you would be fine with that?

Reply:
According to the spec, parsers must be comfortable with that 😄 Also it makes me comfortable to 😄